mne/io/cnt/cnt.py

```diff
@@ -335,12 +336,23 @@ def _get_cnt_info(input_fname, eog, ecg, emg, misc, data_format, date_format, he
                 )
                 n_bytes = 2
                 n_samples = data_size // (n_bytes * n_channels)
+                annotations = _read_annotations_cnt(input_fname, data_format="int16")
+                if annotations.onset[-1] * sfreq > n_samples:
+                    n_bytes = 4
+                    n_samples = n_samples_header
+                    warn(
+                        "Annotations are outside data range. "
+                        "Changing data format to 'int32'."
+                    )
```

Comment:
@larsoner I added a check to see if there were any annotations outside the actual amount of data. If there are, I switch the data_format. I could not figure out a way to detect the correct number of samples with certainty. It only triggers when using the `'auto'` data_format which already doesn't work with the current testing data I uploaded. Is this fine with you?